### PR TITLE
122 update angular models

### PR DESCRIPTION
**Pull request by Shojiki-Lukhozi**
# Description

Updated the Angular Models based on revised diagrams. Added and updated create page to conform to models.

Fixes # 122

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/app/ingredient/utils/src/interfaces/ingredient.interface.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 export interface IIngredient {
   ingredientId?: string;
```
Comment (ryanbasiltrickett) [inline libs/app/ingredient/utils/src/interfaces/ingredient.interface.ts:2]: Please follow ingredient model

File: libs/app/ingredient/utils/src/interfaces/ingredient.interface.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
...
   name: string;
+  unit: string;
+  amount: number;
   tags?: string[];
```
Comment (ryanbasiltrickett) [inline libs/app/ingredient/utils/src/interfaces/ingredient.interface.ts:6]: Please follow ingredient model

File: libs/app/meal-plan/utils/src/interfaces/meal-plan.interface.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { IRecipeDesc } from '@fridge-to-plate/app/recipe/utils';
+
+export interface IMealPlan {
+  mealplanId?: string;
```
Comment (ryanbasiltrickett) [inline libs/app/meal-plan/utils/src/interfaces/meal-plan.interface.ts:4]: Please follow meal plan model

File: libs/app/meal-plan/utils/src/interfaces/meal-plan.interface.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  username: string;
+  date: Date;
+  Breakfast: IRecipeDesc | null;
```
Comment (ryanbasiltrickett) [inline libs/app/meal-plan/utils/src/interfaces/meal-plan.interface.ts:7]: Please follow meal plan model

File: libs/app/meal-plan/utils/src/interfaces/meal-plan.interface.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  Lunch: IRecipeDesc | null;
```
Comment (ryanbasiltrickett) [inline libs/app/meal-plan/utils/src/interfaces/meal-plan.interface.ts:8]: Please follow meal plan model

File: libs/app/meal-plan/utils/src/interfaces/meal-plan.interface.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  Dinner: IRecipeDesc | null;
```
Comment (ryanbasiltrickett) [inline libs/app/meal-plan/utils/src/interfaces/meal-plan.interface.ts:9]: Please follow meal plan model

File: libs/app/meal-plan/utils/src/interfaces/meal-plan.interface.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+  Snack: IRecipeDesc | null;
```
Comment (ryanbasiltrickett) [inline libs/app/meal-plan/utils/src/interfaces/meal-plan.interface.ts:10]: Please follow meal plan model

File: libs/app/meal-plan/utils/src/interfaces/meal-plan.interface.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+  Dessert: IRecipeDesc | null;
```
Comment (ryanbasiltrickett) [inline libs/app/meal-plan/utils/src/interfaces/meal-plan.interface.ts:11]: Please follow meal plan model

File: libs/app/preference/utils/src/interfaces/preference.interface.ts [hunk 9]
```diff
@@ -0,0 +1,8 @@
+export interface IPreference {
+  userId?: string;
```
Comment (ryanbasiltrickett) [inline libs/app/preference/utils/src/interfaces/preference.interface.ts:2]: Please follow preferences model

File: libs/app/preference/utils/src/interfaces/preference.interface.ts [hunk 10]
```diff
@@ -0,0 +1,8 @@
+export interface IPreference {
```
Comment (ryanbasiltrickett) [inline libs/app/preference/utils/src/interfaces/preference.interface.ts:1]: Please follow preferences model naming

File: libs/app/preference/utils/src/interfaces/preference.interface.ts [hunk 11]
```diff
@@ -0,0 +1,8 @@
...
+  userId?: string;
+  username: string;
+  darkMode: boolean;
+  recommendNotification: boolean;
```
Comment (ryanbasiltrickett) [inline libs/app/preference/utils/src/interfaces/preference.interface.ts:5]: Please follow preferences model

File: libs/app/preference/utils/src/interfaces/preference.interface.ts [hunk 12]
```diff
@@ -0,0 +1,8 @@
...
+  viewsNotification: boolean;
```
Comment (ryanbasiltrickett) [inline libs/app/preference/utils/src/interfaces/preference.interface.ts:6]: Please follow preferences model

File: libs/app/preference/utils/src/interfaces/preference.interface.ts [hunk 13]
```diff
@@ -0,0 +1,8 @@
...
+  reviewsdNotification: boolean;
```
Comment (ryanbasiltrickett) [inline libs/app/preference/utils/src/interfaces/preference.interface.ts:7]: Please follow preferences model

File: libs/app/profile/utils/src/interfaces/profile.interface.ts [hunk 14]
```diff
@@ -1,26 +1,17 @@
 import { IIngredient } from '@fridge-to-plate/app/ingredient/utils';
-import { IRecipe } from '@fridge-to-plate/app/recipe/utils';
+import { IRecipeDesc } from '@fridge-to-plate/app/recipe/utils';
+import { IPreference } from '@fridge-to-plate/app/preference/utils';
+import { IMealPlan } from '@fridge-to-plate/app/meal-plan/utils';
 
 export interface IProfile {
-    profileId: string;
-    displayName: string;
+    profileId?: string;
```
Comment (ryanbasiltrickett) [inline libs/app/profile/utils/src/interfaces/profile.interface.ts:7]: Please follow profile model

File: libs/app/profile/utils/src/interfaces/profile.interface.ts [hunk 15]
```diff
@@ -1,26 +1,17 @@
...
-        viewsNotifi: boolean;
-        reviewNotifi: boolean;
-    }
-    mealPlan: {
-        breakfast: IRecipe | null;
-        lunch: IRecipe | null;
-        dinner: IRecipe | null;
-        snack: IRecipe | null;
-    } | null;
+    currMealPlan: IMealPlan[] | null;
```
Comment (ryanbasiltrickett) [inline libs/app/profile/utils/src/interfaces/profile.interface.ts:13]: Please follow profile model

File: libs/app/profile/utils/src/interfaces/profile.interface.ts [hunk 16]
```diff
@@ -1,26 +1,17 @@
...
+    preferences: IPreference[];
```
Comment (ryanbasiltrickett) [inline libs/app/profile/utils/src/interfaces/profile.interface.ts:14]: Please follow profile model

File: libs/app/recipe/utils/src/interfaces/recipe.interface.ts [hunk 17]
```diff
@@ -1,15 +1,21 @@
...
   recipeId?: string;
   name: string;
-  recipeImage: string;
-  ingredients: IIngredient[];
-  instructions: IRecipeStep[];
-  rating?: number;
-  difficulty: 'Easy' | 'Medium' | 'Hard';
-  prepTime?: number;
-  numberOfServings?: number;
   tags?: string[];
```
Comment (ryanbasiltrickett) [inline libs/app/recipe/utils/src/interfaces/recipe.interface.ts:7]: Please follow recipe description model, shouldn't be optional

File: libs/app/recipe/utils/src/interfaces/recipe.interface.ts [hunk 18]
```diff
@@ -1,15 +1,21 @@
...
+  difficulty: 'Easy' | 'Medium' | 'Hard';
+  recipeImage: string;
 }
+
+export interface IRecipe extends IRecipeDesc {
+  description: string;
```
Comment (ryanbasiltrickett) [inline libs/app/recipe/utils/src/interfaces/recipe.interface.ts:13]: Please follow recipe model

File: libs/app/recipe/utils/src/interfaces/recipe.interface.ts [hunk 19]
```diff
@@ -1,15 +1,21 @@
...
+  numberOfServings: number;
```
Comment (ryanbasiltrickett) [inline libs/app/recipe/utils/src/interfaces/recipe.interface.ts:14]: Please follow recipe model

File: libs/app/review/utils/src/interfaces/review.interface.ts [hunk 20]
```diff
@@ -0,0 +1,7 @@
+export interface IReview {
+  reviewId?: string;
+  recipeId?: string;
```
Comment (ryanbasiltrickett) [inline libs/app/review/utils/src/interfaces/review.interface.ts:3]: Please follow review model, shouldn't be optional

**Review by ryanbasiltrickett — CHANGES_REQUESTED** [review]
Please see all requested changes

**Comment by ryanbasiltrickett** [comment]
Please add specified tests under the test section of the PR, and add labels, milestone, etc

**Review by simphiwe-nonabe — APPROVED** [review]
Well done, my boy🦥

**Pull request merged**